### DISCUSSION TITLE
Neper installation error on a cluster

### DISCUSSION THREAD

**Discussion by Maohua10** [neperfepx/neper#619] [Q&A] [answered]
Hi all, I am a brand new rookie in this field and your kind suggestions/help are greatly appreciated, thank you in advance! 
 
I was trying to install Neper for my user on our uinversity's cluster. As instructed in the documentaion, I received the following messages after `cmake ..`

```
-- The C compiler identification is Intel 19.1.2.20200623
-- The CXX compiler identification is Intel 19.1.2.20200623
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/aci/intel/compilers_and_libraries_2020.2.254/linux/bin/intel64/icc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/aci/intel/compilers_and_libraries_2020.2.254/linux/bin/intel64/icpc - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenMP_C: -qopenmp (found version "5.0") 
-- Found OpenMP_CXX: -qopenmp (found version "5.0") 
-- Found OpenMP: TRUE (found version "5.0")  
-- Could NOT find system Scotch - using built-in
-- Could NOT find system NLopt - using built-in
-- Neper is set to be installed in: /usr/local/bin and /usr/share
-- Configuring done
CMake Error at neper_t/CMakeLists.txt:31 (add_library):
  Cannot find source file:

    ./net_tess/net_tess_opt/net_tess_opt_comp/net_tess_opt_comp_objective/net_tess_opt_comp_objective_fval/net_tess_opt_comp_objective_fval_gen/net_tess_opt_comp_objective_fval_gen_stat/net_tess_opt_comp_objective_fval_gen_stat1.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx


CMake Error at neper_t/CMakeLists.txt:31 (add_library):
  No SOURCES given to target: neper_t


CMake Generate step failed.  Build files cannot be regenerated correctly.
```
I tried to search for similar discussions but did not find any...What should I do with the error messages? 
Since I am literally a fresher, it would be appreciated if one can share some common guidance with me for solving similar issues. My current does not allow me to disassemble the problems and I have no idea where to go.

Thank you very much in advance,
Maohua

**Comment by rquey** [comment, marked as answer]
The only unusual thing I see is that you are using an Intel compiler. I've never tried this (but it probably *does not* work).

It does not explain the error you are getting *per se* (and which remains obscure to me), but it is my only clue.

**Reply by mkasemer** [reply]
I am also unfamiliar with the specific error, but I can confirm that I have had issues in the past attempting to compile via Intel compilers on various HPC clusters, and have in the past always resorted to building GNU (except on Mac, which had its own specific challenges/solutions).

**Reply by Maohua10** [reply]
Thanks for the very helpful hints, mkasemer!

**Reply by Maohua10** [reply]
> The only unusual thing I see is that you are using an Intel compiler. I've never tried this (but it probably _does not_ work).
> 
> It does not explain the error you are getting _per se_ (and which remains obscure to me), but it is my only clue.

Thank you Dr. Quey for your kind response!!
